File: src/main/java/appbuilder/api/classes/ConnectionFactory.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package appbuilder.api.classes;

import appbuilder.api.methods.MethodBuilder;
import appbuilder.api.vars.AttributeBuilder;

/**
 *
 * @author psilva
 */
public class ConnectionFactory extends ClassBuilder {

    public ConnectionFactory(String pacote, String caminho) {
        super("ConnectionFactory", pacote, caminho);
        addImport("java.sql.SQLException");
        addImport("java.sql.DriverManager");
        addImport("java.sql.Connection");

        AttributeBuilder user = new AttributeBuilder("private", "String", "user");
        AttributeBuilder password = new AttributeBuilder("private", "String", "password");
        AttributeBuilder database = new AttributeBuilder("private", "String", "database");
        AttributeBuilder host = new AttributeBuilder("private", "String", "host");

        user.addModificador("static");
        password.addModificador("static");
        database.addModificador("static");
        host.addModificador("static");
        addAttribute(user);
        addAttribute(password);
        addAttribute(database);
        addAttribute(host);

        //inicializar
        //os atributos com seus devidos valores padrões
        user.ativarInicialização("\"root\"");
        password.ativarInicialização("\"root\"");
        database.ativarInicialização("\"project\"");
        host.ativarInicialização("\"localhost\"");

        MethodBuilder getConnection = new MethodBuilder("public", "static", "Connection", "getConnection");
        getConnection.addException((ExceptionBuilder) getClasse("SQLException"));
        getConnection.setReturn(getClasse("DriverManager").
                callStatic("getConnection",
                        "\"jdbc:mysql://\"+" + host.getReferencia() + "+\"/\"+" + database.getReferencia() + "",
                        user.getReferencia(), password.getReferencia()));

        addMethod(getConnection);
    }

    public void setServidor(String servidor) {
        getAttribute("host").setValor("\"" + servidor + "\"");
    }

    public String getServidor() {
        return getAttribute("host").getValor().replace("\"", "");
    }

    public void setSenha(String senha) {
        getAttribute("password").setValor("\"" + senha + "\"");
    }

    public String getSenha() {
        return getAttribute("password").getValor().replace("\"", "");
    }

    public void setBaseDeDados(String base) {
        getAttribute("database").setValor("\"" + base + "\"");
    }

    public String getBaseDeDados() {
        return getAttribute("database").getValor().replace("\"", "");
    }

    public void setUsuário(String usuario) {
        getAttribute("user").setValor("\"" + usuario + "\"");
    }

    public String getUsuário() {
        return getAttribute("user").getValor().replace("\"", "");
    }

}

File: AppBuilder/AppBuilder/Old/Classes/ConnectionFactory.cs
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

namespace Api.Old.Classes;

/// <summary>
/// </summary>
/// <remarks>@authorpsilva</remarks>
public class ConnectionFactory : ClassBuilder
{
    public ConnectionFactory(string pacote, string caminho) : base("ConnectionFactory", pacote, caminho)
    {
        AddImport("java.sql.SQLException");
        AddImport("java.sql.DriverManager");
        AddImport("java.sql.Connection");
        AttributeBuilder user = new AttributeBuilder("private", "String", "user");
        AttributeBuilder password = new AttributeBuilder("private", "String", "password");
        AttributeBuilder database = new AttributeBuilder("private", "String", "database");
        AttributeBuilder host = new AttributeBuilder("private", "String", "host");
        user.AddModificador("static");
        password.AddModificador("static");
        database.AddModificador("static");
        host.AddModificador("static");
        AddAttribute(user);
        AddAttribute(password);
        AddAttribute(database);
        AddAttribute(host);

        //inicializar
        //os atributos com seus devidos valores padrões
        user.AtivarInicialização("\\\"root\\\"");
        password.AtivarInicialização("\\\"root\\\"");
        database.AtivarInicialização("\\\"project\\\"");
        host.AtivarInicialização("\\\"localhost\\\"");
        MethodBuilder getConnection = new MethodBuilder("public", "static", "Connection", "getConnection");
        getConnection.AddException((ExceptionBuilder)GetClasse("SQLException"));
        getConnection.SetReturn(GetClasse("DriverManager").CallStatic("getConnection",
            "\\\"jdbc:mysql://\\\"+" + host.GetReferencia() + "+\\\"/\\\"+" + database.GetReferencia() + "",
            user.GetReferencia(), password.GetReferencia()));
        AddMethod(getConnection);
    }

    public virtual void SetServidor(string servidor)
    {
        GetAttribute("host").SetValor("\\\"" + servidor + "\\\"");
    }

    public virtual string GetServidor()
    {
        return GetAttribute("host").GetValor().Replace("\\\"", "");
    }

    public virtual void SetSenha(string senha)
    {
        GetAttribute("password").SetValor("\\\"" + senha + "\\\"");
    }

    public virtual string GetSenha()
    {
        return GetAttribute("password").GetValor().Replace("\\\"", "");
    }

    public virtual void SetBaseDeDados(string @base)
    {
        GetAttribute("database").SetValor("\\\"" + @base + "\\\"");
    }

    public virtual string GetBaseDeDados()
    {
        return GetAttribute("database").GetValor().Replace("\\\"", "");
    }

    public virtual void SetUsuário(string usuario)
    {
        GetAttribute("user").SetValor("\\\"" + usuario + "\\\"");
    }

    public virtual string GetUsuário()
    {
        return GetAttribute("user").GetValor().Replace("\\\"", "");
    }
}